src/main/java/org/openmaptiles/layers/Transportation.java

```diff
@@ -245,6 +255,40 @@ private static boolean isResidentialOrUnclassified(String highway) {
     return "residential".equals(highway) || "unclassified".equals(highway);
   }
 
+  private static boolean isTrunkForZ5(String highway, List<RouteRelation> routeRelations) {
+    // Allow trunk roads that are part of a nation's most important route network to show at z5
+    if (!"trunk".equals(highway)) {
+      return false;
+    }
+    return routeRelations.stream()
+      .map(RouteRelation::networkType)
+      .filter(Objects::nonNull)
+      .anyMatch(Z5_TRUNK_BY_NETWORK::contains);
+  }
+
+  private static boolean isMotorwayWithNetworkForZ4(List<RouteRelation> routeRelations) {
```

Comment:
should this be named `isTrunkWithNetworkForZ4` ? Or should we also run this check down below:

```java
        case FieldValues.CLASS_MOTORWAY ->
          isMotorwayForZ4(routeRelations) ||
          isMotorwayWithNetworkForZ4(routeRelations) ?
          MINZOOMS.getOrDefault(FieldValues.CLASS_MOTORWAY, Integer.MAX_VALUE) : 5;
```